### PR TITLE
Test: SentinelOne MC-1173 login with api token, MC-642 - add 2 unit tests

### PR DESCRIPTION
**Pull request by blaxminarayan-r7**
### Description

Describe the proposed changes:

  - Add option to authentication with API key

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [ ] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [ ] Screenshot of job output with the plugin changes
- [ ] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

Assessments have to many lines. If it's needed  i will send it on slack

### CODE REVIEW

File: plugins/sentinelone/help.md [hunk 1]
```diff
@@ -30,14 +30,18 @@ The connection configuration accepts the following parameters:
 
 |Name|Type|Default|Required|Description|Enum|Example|
 |----|----|-------|--------|-----------|----|-------|
-|credentials|credential_username_password|None|True|Username and password|None|{"username": "user@example.com", "password": "mypassword"}|
-|url|string|None|True|SentinelOne Console URL|None|https://example.sentinelone.com|
+|api_key|credential_secret_key|None|False|Credential secret API key. Provide if you choose API Token Auth type|None|9de5069c5afe602b2ea0a04b66beb2c0|
+|authentication_type|string|Basic Auth|True|Type of authentication|['Basic Auth', 'API Token Auth']|Basic Auth|
+|basic_auth_credentials|credential_username_password|None|False|Username and password. Provide if you choose Basic Auth type|None|{"username": "https://example.com", "password": "mypassword"}|
```
Comment (dsliwinski-r7) [inline plugins/sentinelone/help.md:35]:
```suggestion
|basic_auth_credentials|credential_username_password|None|False|Username and password. Provide if you choose Basic Auth type|None|{"username": "user@example.com", "password": "mypassword"}|
```

File: plugins/sentinelone/help.md [hunk 2]
```diff
@@ -567,8 +571,8 @@ This action is used to start a Deep Visibility Query and get the queryId. You ca
 |Name|Type|Default|Required|Description|Enum|Example|
 |----|----|-------|--------|-----------|----|-------|
 |account_ids|[]string|None|False|List of account IDs to filter by|None|["225494730938491234", "225494730938491235"]|
-|from_date|string|None|True|From date|None|2021-03-01 04:49:26.257525|
-|to_date|string|None|True|Events created before or at this timestamp|None|2021-03-20 04:49:26.257525|
+|from_date|string|None|True|From date|None|2021-03-01 04:49:26.257525+00:00|
+|to_date|string|None|True|Events created before or at this timestamp|None|2021-03-20 04:49:26.257525+00:00|
```
Comment (dsliwinski-r7) [inline plugins/sentinelone/help.md:575]: We should restore these examples :) 

```suggestion
|from_date|string|None|True|From date|None|2021-03-01 04:49:26.257525|
|to_date|string|None|True|Events created before or at this timestamp|None|2021-03-20 04:49:26.257525|
```

**Review by dsliwinski-r7 — CHANGES_REQUESTED** [review]

File: plugins/sentinelone/help.md [hunk 3]
```diff
@@ -17,6 +17,11 @@ This plugin utilizes the SentinelOne API, the documentation is located in the Se
 
 * Sentinel one API administrative credentials
 
+# Supported Product Versions
+
+* 2.0.0
+* 2.1.0
```
Comment (mgrandys-r7) [inline plugins/sentinelone/help.md:23]: Do we need both versions? Not only the newer one?

**Review by mgrandys-r7 — APPROVED** [review]
Looks good, I have only one question.

File: plugins/sentinelone/help.md (hunk 3 above)
Comment (akarolik-r7) [reply plugins/sentinelone/help.md:23]: No we need two because 2.0.0 have v1 api and 2.1.0 v2 api

**Review by mc-dsliwinski — APPROVED** [review]

**Review by dsliwinski-r7 — APPROVED** [review]

**Comment by mgrandys-r7** [comment]
@blaxminarayan-r7 Unit tests and Plugin Validator are not throwing any error when run locally. Could you please investigate the issue?

**Comment by mgrandys-r7** [comment]
Changes from this PR:
https://github.com/rapid7/insightconnect-plugins/pull/1319
were added here 6 days ago

**Pull request merged**